core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -764,9 +783,16 @@ class BrokerServer(
 
       if (socketServer != null)
         CoreUtils.swallow(socketServer.shutdown(), this)
+
       Utils.closeQuietly(brokerTopicStats, "broker topic stats")
       Utils.closeQuietly(sharePartitionManager, "share partition manager")
 
+      if (persister != null)
+        CoreUtils.swallow(persister.stop(), this)
```

Comment:
This is already called in `SharePartitionManager.close()`.

Reply:
Yes, thats true - but since there are multiple items being closed in `SharePartitionManager` and they are not protected with a try - catch, it can happen that they error out before closing the persister and we may end up with an unexpected thread. 
This was apparent in tests.

Reply:
Will remove this from SharePartitionManager - it should not be closing this is an external dep for it.